### ISSUE TITLE
Query: Incorrect table select for star projection in subquery

### ISSUE THREAD

**Issue by mstaessen** [dotnet/efcore#6380]
### Steps to reproduce

See reproduction case: https://github.com/mstaessen/EFDistinctLinqToSql
### The issue

From all the results of a join table (M-N relationships, modeled as two 1-N relationships), I want to get all distinct results from one side of the join table. Generated SQL contains Disctinct clause but for wrong table.

Observed SQL:

```
SELECT [t].[Id], [t].[Source], [t].[ForeignId], [t].[CreationDate], [t].[LastUpdate], [t].[RemovalDate], [t].[Details], [t].[IsRemoved]
FROM (
    SELECT DISTINCT [x.QueryRun.QueryDefinition0].*
    FROM [QueryRunResult] AS [x0]
    INNER JOIN [Property] AS [x.Property0] ON [x0].[PropertyId] = [x.Property0].[Id]
    INNER JOIN [QueryRun] AS [x.QueryRun0] ON [x0].[QueryRunId] = [x.QueryRun0].[Id]
    INNER JOIN [QueryDefinition] AS [x.QueryRun.QueryDefinition0] ON [x.QueryRun0].[QueryDefinitionId] = [x.QueryRun.QueryDefinition0].[Id]
    WHERE [x.QueryRun.QueryDefinition0].[UserId] = @__userId_0
) AS [t]
ORDER BY [t].[CreationDate] DESC
```

Expected SQL: 

```
SELECT [t].[Id], [t].[Source], [t].[ForeignId], [t].[CreationDate], [t].[LastUpdate], [t].[RemovalDate], [t].[Details], [t].[IsRemoved]
FROM (
    SELECT DISTINCT [x.Property0].*
    FROM [QueryRunResult] AS [x0]
    INNER JOIN [Property] AS [x.Property0] ON [x0].[PropertyId] = [x.Property0].[Id]
    INNER JOIN [QueryRun] AS [x.QueryRun0] ON [x0].[QueryRunId] = [x.QueryRun0].[Id]
    INNER JOIN [QueryDefinition] AS [x.QueryRun.QueryDefinition0] ON [x.QueryRun0].[QueryDefinitionId] = [x.QueryRun.QueryDefinition0].[Id]
    WHERE [x.QueryRun.QueryDefinition0].[UserId] = @__userId_0
) AS [t]
ORDER BY [t].[CreationDate] DESC
```

Exception: 

Invalid column name 'Source'.
Invalid column name 'ForeignId'.
Invalid column name 'CreationDate'.
Invalid column name 'LastUpdate'.
Invalid column name 'RemovalDate'.
Invalid column name 'Details'.
Invalid column name 'IsRemoved'.
Invalid column name 'CreationDate'.

```
   at System.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__107_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.<ExecuteAsync>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncQueryingEnumerable.AsyncEnumerator.<MoveNext>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.<MoveNext>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.<ToListAsync>d__129`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at LinqToSqlWrongDistinctClauseReproduction.Program.<MainAsync>d__1.MoveNext() in C:\Users\MSTAESSE\Source\LinqToSqlWrongDistinctClauseReproduction\src\LinqToSqlWrongDistinctClauseReproduction\Program.cs:line 29
```
### Further technical details

EF Core version: 1.0.0
Operating system: Windows 10
Visual Studio version: VS 2015

**Comment by maumar** [comment]
Simplified repro:

```cs
    public class Program
    {
        public static void Main(string[] args)
        {
            using (var context = new MyContext())
            {
                context.Database.EnsureDeleted();
                context.Database.EnsureCreated();

                var query = context.RunResults
                    .Where(x => x.QueryRun.QueryDefinitionId == 1729)
                    .Select(x => x.Property)
                    .Distinct()
                    .Select(x => x.IsRemoved)
                    .ToList();
            }
        }
    }

    public class MyContext : DbContext
    {
        public DbSet<QueryRun> Runs { get; set; }
        public DbSet<QueryRunResult> RunResults { get; set; }
        public DbSet<Property> Properties { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlServer("Server=.;Database=Repro6380;Trusted_Connection=True;MultipleActiveResultSets=true");
        }
    }

    public class Property
    {
        public int Id { get; protected set; }

        public DateTime CreationDate { get; set; }
        public bool IsRemoved { get; set; }
    }

    public class QueryRun
    {
        public int Id { get; set; }

        public int QueryDefinitionId { get; set; }

        public virtual ICollection<QueryRunResult> Results { get; set; }
    }

    public class QueryRunResult
    {
        public int Id { get; set; }

        public int PropertyId { get; set; }
        public virtual Property Property { get; set; }

        public int QueryRunId { get; set; }
        public virtual QueryRun QueryRun { get; set; }
    }
```

**Comment by smitpatel** [comment]
Being fixed by #6784 
Generated query 
```SQL
SELECT [t].[IsRemoved]
FROM (
    SELECT DISTINCT [x.Property0].*
    FROM [RunResults] AS [x0]
    INNER JOIN [Properties] AS [x.Property0] ON [x0].[PropertyId] = [x.Property0].[Id]
    INNER JOIN [Runs] AS [x.QueryRun0] ON [x0].[QueryRunId] = [x.QueryRun0].[Id]
    WHERE [x.QueryRun0].[QueryDefinitionId] = 1729
) AS [t]
```

**Comment by smitpatel** [comment]
dupe of #6657
